Respond 400 instead of 500 when first header field line starts with SP or HTAB

kenballus opened the pull request:
❓ **What kind of change does this PR introduce?**

* [x] 🐞 bug fix
* [ ] 🐣 feature
* [ ] 📋 docs update
* [ ] 📋 tests/coverage improvement
* [ ] 📋 refactoring
* [ ] 💥 other

📋 **What is the related issue number (starting with `#`)**
#728 

❓ **What is the current behavior?** (You can also link to an open issue here)
Cheroot responds 500 when it receives a request in which the first header field line starts with SP or HTAB, due to an `UnboundLocalError`.

❓ **What is the new behavior (if this is a feature change)?**
It responds 400 instead.

📋 **Contribution checklist:**

(If you're a first-timer, check out
[this guide on making great pull requests][making a lovely PR])

* [X] I wrote descriptive pull request text above
* [X] I think the code is well written
* [X] I wrote [good commit messages]
* [x] I have [squashed related commits together][related squash] after
      the changes have been approved
* [X] Unit tests for the changes exist
* [X] Integration tests for the changes exist (if applicable)
* [X] I used the same coding conventions as the rest of the project
* [x] The new code doesn't generate linter offenses
* [x] Documentation reflects the changes
* [X] The PR relates to *only* one subject with a clear title
      and description in grammatically correct, complete sentences

[good commit messages]: http://chris.beams.io/posts/git-commit/
[making a lovely PR]: https://mtlynch.io/code-review-love/
[related squash]:
https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cherrypy/cheroot/729)
<!-- Reviewable:end -->

kenballus commented on docs/conf.py:
```diff
@@ -72,58 +72,58 @@
-    r"/python/cpython/blob/c39b52f/Lib/poplib\.py#user-content-L297-L302",
-    r"^https://matrix\.to/#",  # these render fully on front-end from anchors
+    r'https://github\.com' r'/python/cpython/blob/c39b52f/Lib/poplib\.py#L297-L302',
+    r'https://github\.com'
+    r'/python/cpython/blob/c39b52f/Lib/poplib\.py#user-content-L297-L302',
```
This is AI slop.

kenballus commented on cheroot/server.py:
```diff
@@ -503,8 +506,8 @@
-                "Bad chunked transfer size: {chunk_size!r}".format(
-                    chunk_size=chunk_size
+                'Bad chunked transfer size: {chunk_size!r}'.format(
+                    chunk_size=chunk_size,
```
Slop. This part of the PR came from this project's own CI. You added this code, not me.

avinashkamat48 reviewed (commented):
The functional change here is hard to review because the PR also reformats a large part of the project from double quotes to single quotes. The title describes a specific HTTP parser behavior change, but the diff includes 30+ files and roughly 1.5k lines of formatting churn, which makes it easy to miss the actual 400-vs-500 logic and creates avoidable merge/blame noise. Could the formatting-only changes be split out or reverted in this PR so the behavioral fix and its tests are reviewable on their own?

kenballus committed:
Catch when the first header line begins with whitespace

When the first header field line begins with whitespace, cheroot
responds 500 due to an UnboundLocalError. This patch checks that at
least one header key has been received before processing folded lines.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
